### PR TITLE
Can use hpk files when upgrading with legacy upgrader.

### PR DESCRIPTION
**Pull request by jontore**
1) HPK files supported for legacy upgrader.
2) When upgrading from non HPK supported camera to HPK it makes sure that it ends up going through a real HPK upgrade.

### CODE REVIEW

File: src/components/device/boxfish.ts [hunk 1]
```diff
@@ -11,8 +11,9 @@ import Locksmith from './../locksmith';
 import CameraEvents from './../../utilitis/events';
 import Detector from './../detector';
 import { EventEmitter } from 'events';
-import createBoxfishUpgraderFactory from './../upgrader/boxfishUpgraderFactory';
+import { createBoxfishUpgrader } from './../upgrader/boxfishUpgraderFactory';
 
+const MAX_UPGRADE_ATTEMT = 3;
```
Comment (kristofferkoch) [inline src/components/device/boxfish.ts]: Is this legacy + hpk upgrade + hpk verify?

File: src/components/upgrader/boxfishUpgrader.ts [hunk 2]
```diff
@@ -310,24 +314,25 @@ export default class BoxfishUpgrader extends EventEmitter implements IDeviceUpgr
     return this._cameraManager.api.setProductInfo({ flash_boot_state: state });
   }
 
-  async initFlash(fsblMvcmd: Buffer, boxfishPkg: BoxfishPackage): Promise<any> {
+  async initFlash(fsblMvcmd: Buffer, boxfishUpgraderFile: IBoxfishUpgraderFile): Promise<any> {
```
Comment (kristofferkoch) [inline src/components/upgrader/boxfishUpgrader.ts]: Is initFlash actually used? Probably not relevant for this review, but you're changing it, so I guess you own it?

File: src/components/upgrader/boxfishUpgrader.ts [hunk 3]
```diff
@@ -399,4 +404,26 @@ export default class BoxfishUpgrader extends EventEmitter implements IDeviceUpgr
...
+    const prodInfo = await this._cameraManager.api.getProductInfo();
+    const currentSwVersion = this.extractSofwareVersionFromProdInfo(prodInfo);
+    if (semver.lt(currentSwVersion, HPK_SUPPORT_VERSION)) {
+      return true;
+    }
+
+    try {
+      const response = await this._cameraManager.api.sendAndReceiveMessagePack('',
+        {
+          send: 'camera/upgrade_status',
```
Comment (kristofferkoch) [inline src/components/upgrader/boxfishUpgrader.ts]: Yes, trying to merge the change into `camera/get_state` in kubicam-sw now

File: src/components/upgrader/boxfishUpgraderFactory.ts [hunk 4]
```diff
@@ -5,10 +5,14 @@ import IDeviceUpgrader from './../../interfaces/IDeviceUpgrader';
...
 import HPKUpgrader from './../upgrader/hpkUpgrader';
 
+import BoxfishPkg from './boxfishpkg';
+import BoxfishHpk from './boxfishhpk';
+
 import { EventEmitter } from 'events';
 
-const HPK_SUPPORT_VERSION = '1.1.0-0';
-export default async function createBoxfishUpgrader(
+export const HPK_SUPPORT_VERSION = '1.1.0-0';
```
Comment (kristofferkoch) [inline src/components/upgrader/boxfishUpgraderFactory.ts]: We should probably bump kubicam-sw version to 1.2.0. The interface has been extended (camera/get_state).

File: src/components/device/boxfish.ts (hunk 1 above)
Comment (jontore) [reply src/components/device/boxfish.ts]: No it is allowed 3 attempts if the status after upgrade is invalid.

File: src/components/device/huddlygo.ts [hunk 5]
```diff
@@ -251,4 +251,8 @@ export default class HuddlyGo extends UvcBaseDevice implements IDeviceManager {
   getDetector(): IDetector {
     throw new Error('Detections are not supported on Huddly GO camera!');
   }
+
+  getState(): Promise<any> {
+    throw new Error('State is supported on Huddly GO camera!');
```
Comment (brikendr) [inline src/components/device/huddlygo.ts]: Did you mean to say `State is not supported on Huddly GO camera!`

**Pull request merged**